blockchain/state_processor.go

```diff
@@ -99,3 +102,37 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 
 	return receipts, allLogs, *usedGas, internalTxTraces, processStats, nil
 }
+
+// ProcessParentBlockHash stores the parent block hash in the history storage contract
+// as per EIP-2935.
+func ProcessParentBlockHash(header *types.Header, vmenv *vm.EVM, statedb vm.StateDB, rules params.Rules) error {
```

Comment:
Can we move this function into `consensus/misc/eip2935.go`? as we adopt Initialize(), the 2935 is more like a part of Istanbul consensus engine. Thus having it under consensus directory.

Reply:
The statement `vmenv.Reset(blockchain.NewEVMTxContext(msg, header, vmenv.ChainConfig()), statedb)` matters. `NewEVMTxContext()` is defined in the `blockchain` package, and the `misc` cannot be imported by the cycle.

Of course I can pass it `any` type, but it is fragile if the change is made in the signature of  `NewEVMTxContext` without considering the `ProcessParentBlockHash` function because its dependency is not duty of compiler. Also, adding function signature comparison at runtime, but still not effective method I think.

Reply:
Oh I see. I'm fine with it then. Thank you for thoughts.